### PR TITLE
First pull request.

### PR DESCRIPTION
**Pull request by ikhat**
Just trying this pull request thing out for the first time.

### CODE REVIEW

File: problemsets-mat223/linearalbegra.tex [hunk 1]
```diff
@@ -973,6 +973,13 @@ \section*{Lines and Planes}
 	Let $A$ be the set of points $(x,y)\in\R^2$ such that $y=2x+1$.
 	\begin{parts}
 		\item Describe $A$ using set-builder notation.
+			\begin{solution}
+				$\Set*{\vec v \in \R^2 \given \vec v = \begin{bmatrix*}[c] x \\ 2x+1\end{bmatrix*} \text{ for some } x \in \R}$
```
Comment (siefkenj) [inline problemsets-mat223/linearalbegra.tex]: Use `\mat{...}` instead of `\begin...\end` There should be a `\matc` if you want it centered.

File: problemsets-mat223/linearalbegra.tex [hunk 2]
```diff
@@ -1038,24 +1045,81 @@ \section*{Lines and Planes}
 	$z=y$.
 	\begin{parts}
 		\item Write $\ell$ in vector form. Is vector form of $\ell$ unique?
-		\item Write $L$ in vector form. 
+			\begin{solution}
+				$\vec x = t \mat{1 \\ -2} + \mat{0 \\ 3}$
+
+				The vector form is not unique, as any scalar multiple of the
```
Comment (siefkenj) [inline problemsets-mat223/linearalbegra.tex]: non-zero multiple

File: problemsets-mat223/linearalbegra.tex [hunk 3]
```diff
@@ -1038,24 +1045,81 @@ \section*{Lines and Planes}
...
+				same line: $\vec x = t \mat{-4 \\ 8} + \mat{1 \\ 1}$.
+			\end{solution}
+		\item Write $L$ in vector form.
+			\begin{solution}[inline]
+				$\vec x = t\mat{1 \\ -2 \\ -2} + \mat{0 \\ 3 \\ 3}$
+			\end{solution}
 		\item Find another vector form for $L$ where both ``$\vec d$'' and
 			``$\vec p$'' are different from before.
+			\begin{solution}
+				Again, any multiple of the direction vector will work for
```
Comment (siefkenj) [inline problemsets-mat223/linearalbegra.tex]: non-zero multiple

File: problemsets-mat223/linearalbegra.tex [hunk 4]
```diff
@@ -1038,24 +1045,81 @@ \section*{Lines and Planes}
...
-	\[\overbrace{\vec x=t\mat{1\\2\\3}+\mat{0\\0\\1}}^{\displaystyle A}
+	\[
+	\overbrace{\vec x=t\mat{1\\2\\3}+\mat{0\\0\\1}}^{\displaystyle A}
 	\qquad \overbrace{\vec x=t\mat{-1\\1\\1}+\mat{-1\\1\\2}}^{\displaystyle B}
 	\qquad \overbrace{\vec x=t\mat{2\\-1\\1}+\mat{1\\1\\1}}^{\displaystyle C}.
 	\]
 	\begin{parts}
 		\item Do the lines $A$ and $B$ intersect? Justify your conclusion.
+			\begin{solution}
+				Yes. In this case this is quite easy to see since
```
Comment (siefkenj) [inline problemsets-mat223/linearalbegra.tex]: Don't say "easy to see". I'd like this answer as "Yes, $t\mat{1\\2\\3} + ... = -1\mat ...$. To find the intersection, we solve the vector equation ... for s and t." Or something similar. That is, explicit answer up front, then explanation.

File: problemsets-mat223/linearalbegra.tex [hunk 5]
```diff
@@ -1038,24 +1045,81 @@ \section*{Lines and Planes}
...
+			\end{solution}
 		\item Do the lines $A$ and $C$ intersect? Justify your conclusion.
+			\begin{solution}
+				No. In order for them to intersect, there would have to exist 
+				two real numbers $a$ and $c$ such that when substituded for $t$ 
+				in the equations of $A$ and $C$, respectively, the resulting 
+				vectors would equal. In other words, we would have to have:
+				\[
+					a\mat{1\\2\\3}+\mat{0\\0\\1} = c\mat{2\\-1\\1}+\mat{1\\1\\1}.
+				\]
```
Comment (siefkenj) [inline problemsets-mat223/linearalbegra.tex]: We can make this more consise. "No. The vector equation ... has no solution since.... Therefore the lines cannot intersect"

File: problemsets-mat223/linearalbegra.tex [hunk 6]
```diff
@@ -1038,24 +1045,81 @@ \section*{Lines and Planes}
...
+				into the first equation forces $a = -\tfrac{1}{5}$ and therefore
+				$c = -\tfrac{3}{5}$. These two numbers don't satisfy the second
+				equation, however.
+			\end{solution}
 		\item Let $\vec p\neq \vec q$ and suppose
 			$X$ has vector form $\vec x=t\vec d+\vec p$ and $Y$ has
 			vector form $\vec x=t\vec d+\vec q$. Is it possible
 			that $X$ and $Y$ intersect?
+			\begin{solution}
+				Yes. This can happen for example if $\vec q$ is a point, other
```
Comment (siefkenj) [inline problemsets-mat223/linearalbegra.tex]: How about something like: "Yes. If $\vec q=\vec p+a\vec d$ for $a\neq 0$, then ..." This way we avoid saying "the vector is a point".

File: problemsets-mat223/linearalbegra.tex [hunk 7]
```diff
@@ -1109,25 +1173,93 @@ \section*{Lines and Planes}
...
-	\[\overbrace{\vec x=t\mat{1\\2\\3}+\mat{0\\0\\1}}^{\displaystyle A}
-	\qquad \overbrace{\vec x=t\mat{-1\\1\\1}+\mat{-1\\1\\2}}^{\displaystyle B}.
+	\[
+		\overbrace{\vec x=t\mat{1\\2\\3}+\mat{0\\0\\1}}^{\displaystyle A}
+		\qquad
+		\overbrace{\vec x=t\mat{-1\\1\\1}+\mat{-1\\1\\2}}^{\displaystyle B}.
 	\]
 	Let $\mathcal P$ the plane that contains the lines $A$ and $B$.
 	\begin{parts}
 		\item Find two direction vectors in $\mathcal P$.
```
Comment (siefkenj) [inline problemsets-mat223/linearalbegra.tex]: For these, rewrite so answer is up front and explanation follows.

File: problemsets-mat223/linearalbegra.tex [hunk 8]
```diff
@@ -1179,32 +1311,74 @@ \section*{Span}
 	Let $\ell_1\subseteq \R^2$ be the line with equation $x-y=0$ and $\ell_2\subseteq\R^2$
 	the line with equation $x-y=4$.
 	\begin{parts}
-		\item If possible, describe $\ell_1$ as a span. Otherwise explain why it's not possible.
-		\item If possible, describe $\ell_2$ as a span. Otherwise explain why it's not possible.
+		\item If possible, describe $\ell_1$ as a span. Otherwise explain why 
+			it's not possible.
+			\begin{solution}
+				$\ell_1$ is the span of the vector $\mat{1\\1}$, since a point
```
Comment (siefkenj) [inline problemsets-mat223/linearalbegra.tex]: Write the first sentence as an equation.

File: problemsets-mat223/linearalbegra.tex [hunk 9]
```diff
@@ -1179,32 +1311,74 @@ \section*{Span}
 	Let $\ell_1\subseteq \R^2$ be the line with equation $x-y=0$ and $\ell_2\subseteq\R^2$
 	the line with equation $x-y=4$.
 	\begin{parts}
-		\item If possible, describe $\ell_1$ as a span. Otherwise explain why it's not possible.
-		\item If possible, describe $\ell_2$ as a span. Otherwise explain why it's not possible.
+		\item If possible, describe $\ell_1$ as a span. Otherwise explain why 
+			it's not possible.
+			\begin{solution}
+				$\ell_1$ is the span of the vector $\mat{1\\1}$, since a point 
+				$\mat{1\\1}$ is on this line of and only if $x = y$, which in
```
Comment (siefkenj) [inline problemsets-mat223/linearalbegra.tex]: of -> if

File: problemsets-mat223/linearalbegra.tex [hunk 10]
```diff
@@ -1179,32 +1311,74 @@ \section*{Span}
...
+			\begin{solution}
+				$\ell_1$ is the span of the vector $\mat{1\\1}$, since a point 
+				$\mat{1\\1}$ is on this line of and only if $x = y$, which in 
+				turn is true if and only if it is a scalar multiple of $\mat{1\\1}$. 
+			\end{solution}
+		\item If possible, describe $\ell_2$ as a span. Otherwise explain why it's not 
+			possible.
+			\begin{solution}
+				This is not possible. The origin $\mat{0\\0}$ must always be in the span
+				of any set of vectors, since we can use all zeroes as the scalars in a
```
Comment (siefkenj) [inline problemsets-mat223/linearalbegra.tex]: Too wordy

File: problemsets-mat223/linearalbegra.tex (hunk 1 above)
Comment (ikhat) [reply problemsets-mat223/linearalbegra.tex]: Cool. Didn't know about \matc. Will do.

File: problemsets-mat223/linearalbegra.tex (hunk 6 above)
Comment (siefkenj) [reply problemsets-mat223/linearalbegra.tex]: Yeah. It's yucky...in a vector space there are no points.

In this class we say that every vector corresponds to the point it points to from the origin and every point corresponds to a vector from the origin to that point, so we will use the interchangeably. That's said on day 1 or 2 and it is mostly ignored from there on out.

Comment (ikhat) [reply problemsets-mat223/linearalbegra.tex]: Gotcha. 

There were a couple of other related things I noticed. I saw that you ask them for  "vectors in a plane", for example. It's because of that and a few similar things that I figured I could be more loose with the distinction.

File: problemsets-mat223/linearalbegra.tex [hunk 11]
```diff
@@ -1561,9 +1953,23 @@ \section*{Dot Product}
...
+			\end{solution}
 		\item Does there exists a \emph{unit vector} $\vec x$ that is distance
 			$1$ from $\vec u$?
+			\begin{solution}
+				No. $\|\vec u\| = \sqrt{6}$, and so the shortest length that a
+				vector whose distance from $\vec u$ is 1 can have is $\sqrt{6} - 1$, 
+				which is greater than 1. 
+			\end{solution}
 		\item Suppose $\vec y$ is a unit vector and the distance between $\vec y$ and
 			$\vec u$ is $2$.  What is the angle between $\vec y$ and $\vec u$?
```
Comment (ikhat) [inline problemsets-mat223/linearalbegra.tex]: Quick question about this problem (problem 21.4, line 1973). The solution to this I imagine requires understanding the distributivity of the dot product: 

 4 = (u - v) \cdot (u - v) = u \cdot u - 2 u \cdot v + v \cdot v

(Then use that to compute u \cdot v, from which you can get the angle.)

That isn't something that is explicitly mentioned earlier. Is that something we expect them to discover? If so, to what extent should I be explaining it before using it? 

Or am I just totally missing some more elementary way of doing this?

Comment (siefkenj) [reply problemsets-mat223/linearalbegra.tex]: Yeah, they will know distributivity of the dot product. The unfortunate state of affairs is that for most students it won't even occur to them to ask, "hey, is this new product distributive?" since to them all things are distributive. We don't emphasize axioms in this course, so I'm okay with them just treating the dot product as distributive without proving it or having it as part of the definition.

**Comment by siefkenj** [comment]
This looks pretty good. I'm going to merge this for now, and we can do cleanups in another PR. Are you familiar with branches? It would be nice if the solutions to each problem were in a separate PR which can be achieved with branches.

```
git checkout -b prob44
```

for example will create a new branch. You can write solutions for one problem, push that branch and make a PR from it. Then, checkout `master` again and make a new branch for the next problem. This way we can merge some solutions while continuing to edit others instead of having to wait and doing massive PRs.

**Comment by ikhat** [comment]
I'm familiar with the idea behind branches, but have never worked with
them. I've done a bit of reading. Let me know if this is all correct.

Starting on my master branch, git branch prob44 creates a new branch called
prob44 but keeps you working on whatever branch you were on before. git
checkout -b prob44 creates a new branch called prob44 and checks it out.

I make sure I'm on the prob44 branch, do stuff only with that problem, then
commit the changes as normal to my fork of your repository. Then, on
github, I'll have the option to create a pull request just from my given
branch? That seems simple enough.

What happens with mine when you approve the work I've done on a given
problem and merge that branch into the master of your repository? Do I have
to merge it here as well to keep them synchronized? Or is this automated in
some way?

(Sorry for all the questions, but I'm both new to this and very curious to
learn it.)

**Comment by siefkenj** [comment]
Your description is correct. When I merge your changes, you will have to resync with my branch. https://help.github.com/articles/syncing-a-fork/

**Comment by ikhat** [comment]
Got it. Same procedure as before.

I actually had a quick question about syncing a fork. The first time I did
it (after you added the mathtools stuff) the merge didn't go through as
normal because it would have created a bunch of conflicts. I hadn't done
anything significant though, so I just forced the sync and overwrote my
version with yours and went from there.

In the future, I guess this won't be an issue because I'll be doing all of
my changes on branches, and only syncing my master branch with yours after
you merge my changes. And the act of syncing with your master will also
merge my branch with my master as well?
